### DISCUSSION TITLE
All About Unread E-Mail Notifications

### DISCUSSION THREAD

**Discussion by jcsteele** [joelekstrom/fastmate#60] [Show and tell]
One of the great features Fastmate has is to display unread e-mails as a notification in the dock and status bar.  If installed and working correctly, it should look something like below:

No Unread Mails: Status Icon and Dock
![statusicon-without](https://user-images.githubusercontent.com/5166418/151842797-1adb52bd-6ebd-423f-9d6a-797fa718020d.png)
![dock-without](https://user-images.githubusercontent.com/5166418/151842940-d481fbd5-5ba4-4b61-a6af-2a455c649c19.png)

Unread Mails: Status Icon and Dock
![statusicon-with](https://user-images.githubusercontent.com/5166418/151843004-6a92aee5-d7e4-4982-8dcb-dfaa7758d79c.png)
![dock-with](https://user-images.githubusercontent.com/5166418/151843012-69a61531-8c17-4bda-ae10-fe2ce177b009.png)

If you've noticed that your unread e-mail notifications are not working in the dock or status bar, please check your notification settings within MacOS first for the Fastmate app specific settings.  These can be accessed through System Preferences->Notifications & Focus and ensure "Allow Notifications" is enabled similarly to below:

![NotificationSettings](https://user-images.githubusercontent.com/5166418/151843420-12e2b19e-9f4b-4fc6-8568-706c55cbcefd.png)

Once notifications are enabled, you should start receiving unread e-mail notifications.  Some of the configurable options are below, which can be accessed through the Fastmate application preferences:

![Unreadmailsettings](https://user-images.githubusercontent.com/5166418/151844007-0d083672-3e7e-4bb3-b82f-4925c2ba8adb.png)

**Some Known Limitations**

1. Setting the "For Specific Folders" option to "Archive" has had mixed success.

**Comment by joelekstrom** [comment]
Thanks! Some of the tips here would be nice to add to the main page README.

**Reply by jcsteele** [reply]
How about something like below?

## Enabling Status Bar and Dock Notifications
Upon installation, Fastmate should prompt you to allow notifications.  If this was missed, or you find your status bar and dock notifications are not working, please check the Fastmate-specific notification settings within MacOS by going to System Preferences->Notifications & Focus and ensure that "Allow Notifications" is enabled for Fastmate.  For more information on this features, please see https://github.com/joelekstrom/fastmate/discussions/60

**Reply by joelekstrom** [reply]
Looks good to me! Feel free to submit a PR for the readme

**Comment by u35253** [comment]
Is there any way for me to swap in a different status bar (menu bar) icon for "new mail" in Fastmate?  I am using Fastmate v1.8.0 on Mojave 10.14.6 (maximum updates).

I ask because, in my use case, an alternative new mail icon providing even more obvious differentiation for noticing it would be most welcome.  A screenshot is attached below, comparing with my other menu bar mail apps.

![compare_new_mail_indicators](https://user-images.githubusercontent.com/115374358/215229387-3724a0e2-001e-45a9-b2ff-27d7ff037d5e.png)

If there happens to be a known way for me to swap in an image file on my system in order to use a custom new mail icon, I would probably be able to follow the steps.  Thanks for any ideas!  Fastmate is very nice indeed.

**Reply by joelekstrom** [reply]
Sorry for the slow response - I don't think there's a way to do that easily. Might be able to switch out the image file in the .app, but I'm not sure how that works with code signing.

I'll consider this as a feature request!

**Reply by u35253** [reply]
Okay, thanks.  FWIW, I've moved to macOS Ventura recently, so I am less concerned about making Mojave work any more.

A feature request sounds great.  If Fastmate gets new features for a colored/counting status bar icon when there is unread mail, I'd love to see that trickle in through the auto update checks that are configured.

Thanks very much for the convenient app!
